app/views/index.erb

```diff
@@ -0,0 +1,35 @@
+<body>
+  <br>
+  <h1>Welcome to Chitter</h1>
+  <br>
+  <br>
+  <form class="form-inline" method='post' action='/login'>
+    <legend>Please login here</legend>
+    <div class="form-group">
+      <label for="inputUsername">Username: </label>
+      <input type='text' name='username' class="form-control" placeholder='Enter your username'>
+    </div>
+    <div class="form-group">
+      <label for="inputPassword"> Password: </label>
+      <input type='password' class="form-control" name='password' placeholder='Enter your password'>
+    </div>
+    <button type="submit" class="btn btn-primary" name='Login'>Login</button>
+  </form>
+  <br>
+  <br>
+  <a href='/signup'><button type="button" class="btn btn-primary">Sign up</button></a>
+  <br>
+  <br>
+  <h3>Past peeps: </h3>
+  <br>
+  <br>
+  <% if @public_feed %>
+    <% @public_feed.each do |peep| %>
+      <% user = @all_users.get(peep.user_id) %>
```

Comment:
Use DataMapper to handle this logic: `user = peep.user` - that's what associations are for.